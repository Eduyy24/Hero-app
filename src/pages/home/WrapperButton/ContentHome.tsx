import { FunctionComponent } from "react";
import { GeneralButton } from "../../../components/general-button/GeneralButton";
import styles from "./ContentHome.module.css";

export const ContentHome: FunctionComponent = () => {
  return (
    <div className={styles.container}>
      <GeneralButton name="Comprar" />
    </div>
  );
};
