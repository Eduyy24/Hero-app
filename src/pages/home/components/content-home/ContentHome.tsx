import { FunctionComponent } from 'react';
import { GeneralButton } from '../../../../components/general-button/GeneralButton';
import styles from './ContentHome.module.css';
import {useNavigate} from 'react-router-dom';
import { DESCRIPTION_APP, NAME_APP, NAME_BUTTON_HOME } from '../../../../utils/strings';
import usePagesData from '../../../../hooks/usePagesData';
import heroLogo from '../../../../assets/superhero.png';

export const ContentHome: FunctionComponent = () => {
  const navigate = useNavigate();
  const {getPageForOrder} = usePagesData()
  const nextPage = getPageForOrder(1);

  return (
    <div className={styles.container}>
      <img className={styles.heroLogo} src={heroLogo} alt='hero logo' />
      <h1 className={styles.title}>{NAME_APP}</h1>
      <h2 className={styles.subTitle}>{DESCRIPTION_APP}</h2>
      <div className={styles.wrapperButton}>
        <GeneralButton name={NAME_BUTTON_HOME} onClick={() => {
          navigate(`/${nextPage?.key}`)
        }} />
      </div>
    </div>
  );
};
