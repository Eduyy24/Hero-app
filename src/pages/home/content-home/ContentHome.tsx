import { FunctionComponent } from "react";
import { GeneralButton } from "../../../components/general-button/GeneralButton";
import styles from "./ContentHome.module.css";
import heroLogo from "../../../assets/superhero.png";
import {useNavigate} from 'react-router-dom';

export const ContentHome: FunctionComponent = () => {
  let navigate = useNavigate();
  return (
    <div className={styles.container}>
      <img className={styles.heroLogo} src={heroLogo} alt="hero logo" />
      <h1 className={styles.title}>Hero App</h1>
      <h2 className={styles.subTitle}>Tu heroe de ventas inmobiliario</h2>
      <div className={styles.wrapperButton}>
        <GeneralButton name="Vender" onClick={() => {
          navigate('/user')
        }} />
      </div>
    </div>
  );
};
