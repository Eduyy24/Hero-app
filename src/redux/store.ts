import { configureStore } from '@reduxjs/toolkit'
import thunk from 'redux-thunk';
import storage from 'redux-persist/lib/storage'

import { persistReducer, persistStore } from 'redux-persist';
import generalReducer from './slices/generalSlice'


const persistConfig = {
  key: 'root',
  storage,
}

const persistedReducer = persistReducer(persistConfig, generalReducer)

const store = configureStore({
  reducer: {
    general: persistedReducer
  },
  devTools: process.env.NODE_ENV !== 'production',
  middleware:[thunk]
})

const persistor = persistStore(store);

export { store, persistor };
