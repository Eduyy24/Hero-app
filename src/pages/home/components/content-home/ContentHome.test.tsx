import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../../../redux/store";
import { ContentHome } from "./ContentHome";

describe("<ContentHome />", () => {
  test("render", () => {
    render(
      <Provider store={store}>
        <ContentHome />
      </Provider>
    );

    const content = screen.getByTestId('container-content')
    expect(content).toMatchSnapshot();
  });
});
