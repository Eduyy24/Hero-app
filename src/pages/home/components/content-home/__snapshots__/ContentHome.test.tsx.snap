// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentHome /> render 1`] = `
<div
  class="container"
  data-testid="container-content"
>
  <img
    alt="hero logo"
    class="heroLogo"
    src="superhero.png"
  />
  <h1
    class="title"
  >
    Hero App
  </h1>
  <h2
    class="subTitle"
  >
    Tu heroe de ventas inmobiliario
  </h2>
  <div
    class="wrapperButton"
  >
    <button
      class="generalButton"
      name="Vender"
    >
      Vender
    </button>
  </div>
</div>
`;
